src/transformers/models/musicgen/modeling_musicgen.py

```diff
@@ -0,0 +1,2360 @@
+                **model_kwargs,
+            )
+
+            # 10. run sample
+            outputs = self.sample(
+                input_ids,
+                logits_processor=logits_processor,
+                logits_warper=logits_warper,
+                stopping_criteria=stopping_criteria,
+                pad_token_id=generation_config.pad_token_id,
+                eos_token_id=generation_config.eos_token_id,
+                output_scores=generation_config.output_scores,
+                return_dict_in_generate=generation_config.return_dict_in_generate,
+                synced_gpus=synced_gpus,
+                **model_kwargs,
+            )
+
+        else:
+            raise ValueError("Got incompatible mode for generation, should be one of greedy or sampling.")
+
+        if generation_config.return_dict_in_generate:
+            output_ids = outputs.sequences
+        else:
+            output_ids = outputs
+
+        # build and apply the final delay pattern mask
+        _, pattern_mask = self.build_delay_pattern_mask(
+            inputs, self.generation_config.pad_token_id, max_length=self.generation_config.max_length
+        )
+        output_ids = self.apply_delay_pattern_mask(output_ids, pattern_mask)
+
+        # revert the pattern delay mask by filtering the pad token id
+        output_ids = output_ids[output_ids != self.generation_config.pad_token_id].reshape(
+            batch_size, self.num_codebooks, -1
+        )
+
+        # TODO(SG): hacks to make the output ids compatible with encodec - add to encodec modelling code
+        output_ids = output_ids.unsqueeze(0)
+        audio_scales = [kwargs.get("audio_scales")] * len(output_ids)
+        output_values = self.audio_encoder.decode(
+            output_ids, audio_scales=audio_scales, padding_mask=kwargs.get("padding_mask")
+        )
+
+        if generation_config.return_dict_in_generate:
+            outputs.sequences = output_values
+            return outputs
+        else:
+            return output_values
+
+    def generate_unconditional(self, num_samples=1, **kwargs):
+        """A wrapper around the generate method for generating audio samples without any text or audio prompt,
+        Parameters:
+        enabling the model to be used without the feature extractor or tokenizer.
+            num_samples (int, *optional*):
+                Number of samples to unconditionally generate.
+        """
+        last_hidden_state = torch.zeros(
+            (num_samples, 1, self.config.text_encoder.hidden_size), device=self.device, dtype=self.dtype
+        )
+        encoder_outputs = BaseModelOutput(last_hidden_state=last_hidden_state)
+
+        # TODO(SG): check this works with fp16 precision
+        attention_mask = torch.zeros((num_samples, 1), device=self.device, dtype=torch.long)
+        return self.generate(
+            attention_mask=attention_mask, encoder_outputs=encoder_outputs, guidance_scale=1, **kwargs
+        )
```

Comment:
```suggestion
```

Do we really need this method? Think i'd be nicer if the user would use `generate` for both conditional and unconditional. What do you think about advertising:

```py
encoder_outputs, attention_mask = model.get_null_encoding()
model.generate(encoder_outputs=encoder_outputs, attenton_mask=attention_mask, guidance_scale=1)
```

Think this would be nicer to make sure the user understands that unconditional and conditional generation is the same underlaying function.

Reply:
So if we want to make it easy for the user think it's totally fine to have a `get_null_encoding()` or `get_unconditional_encoding()` but don't think introducing a `generate_unconditional` wrapper is a good idea.

Reply:
That's also good with me - we can write a nice docstring for `get_null_encoding` and showcase on the docs how to pair it with `generate` for unconditional generation 👍

It's then not a 'magic' method at all

Reply:
Resolved in https://github.com/huggingface/transformers/pull/24109/commits/2c7efa4bc4981a762ac0730da89b0f179c1fa405